SYCL/DeviceCodeSplit/aot-accelerator.cpp

```diff
@@ -1,4 +1,5 @@
 // REQUIRES: aoc, accelerator
 
-// RUN: %clangxx -fsycl -fsycl-device-code-split=per_source -fsycl-targets=spir64_fpga -I %S/Inputs -o %t.out %S/split-per-source-main.cpp %S/Inputs/split-per-source-second-file.cpp
+// RUN: %clangxx -fsycl -fsycl-device-code-split=per_source -fsycl-targets=spir64_fpga -I %S/Inputs -o %t.out %S/split-per-source-main.cpp %S/Inputs/split-per-source-second-file.cpp \
```

Comment:
Do we have specific tests for DAE feature?
If yes, should we add multi source test there instead?
These tests are intended to test device code split. I'm not sure we should cover several features by one test.

Reply:
From a quick check, we don't seem to have any dedicated DAE tests. In principle, I agree that burdening a group of tests with several functionality checks is not a good idea. However, in this instance we have an option that is soon to be enabled by default and is known to malfunction with specific features. In essence, the end result with DAE will be similar to device code split, which is implicitly enabled in `auto` mode for other features' tests.

Please let me know what you think.

Reply:
Ok, I got that. In this case I'm ok with this change. It would be great to remove explicit DAE enabling from these tests once it is enabled by default, so it doesn't confuse people.

Reply:
Sure, @mdtoguchi and myself will be tracking the removal once the default enabling is done.

Reply:
> Sure, @mdtoguchi and myself will be tracking the removal once the default enabling is done.

I also wonder whether we need to have some tests specifically for disabled DAE, if such option exists and available to our users

Reply:
The `-fno-*` switch is available for DAE and is actually used in a couple tests, including `SeparateCompile` and `Config/kernel_from_file.cpp`. By the way, these are compiled "by hand" without `sycl-post-link` processing, so enabling DAE for that wouldn't work right now.

I'm not sure if that is sufficient coverage, but on the other hand, would my analogy with `-fsycl-device-code-split=off` not being E2E-tested still stand?

Reply:
> would my analogy with -fsycl-device-code-split=off not being E2E-tested still stand?

So, does that mean we don't have tests that explicitly disable device code split?

Reply:
> So, does that mean we don't have tests that explicitly disable device code split?

@Fznamznon, I wasn't able to find any `-fsycl-device-code-split=off` occurences in the test suite's `RUN` lines. Might have missed, though :D

> I agree that there is a danger of triggering combinatoric explosion, but I don't know whether existing coverage is in good balance or not.

@AlexeySachkov, while the combinatoric explosion in the RUN lines could probably be handled with LIT config magic (e.g. like we do with `%sycl_triple`), we also have to beware of the CI execution time explosion. I'll open a GH issue so that we can transfer our discussion there.

Reply:
> So, does that mean we don't have tests that explicitly disable device code split?

@Fznamznon, I wasn't able to find `-fsycl-device-code-split=off` in any of the test suite `RUN` lines.

> I would consider that as a testing gap/problem as well.

@AlexeySachkov, I've created https://github.com/intel/llvm-test-suite/issues/583 to capture our discussion points there.

> I agree that there is a danger of triggering combinatoric explosion, but I don't know whether existing coverage is in good balance or not.

While triggering the wider set of combinations could be made easier by some LIT config magic (similarly to what we have for `%sycl_triple` and GPU types), explosion of the test execution times is the other thing we have to beware. Captured this in the above issue.